spectator-api/src/main/java/com/netflix/spectator/api/BasicTagFactory.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spectator.api;
+
+import com.netflix.spectator.impl.Preconditions;
+
+/**
+ * Basic implementation of the TagFactory class that does nothing fancy.
+ */
+public final class BasicTagFactory implements TagFactory {
+  private final String name;
+
+  /** Create a new instance that will produce tags with the specified non-null name. */
+  protected BasicTagFactory(String name) {
```

Comment:
I think this is another artifact of an earlier version of the code.  I think that the class should be final, but have a public constructor.  Thinking about this class made me realize that the ConstantTagFactory class that was in the DefaultDynamicId class should really be a standalone, public class.  I can see it being a common way of providing default values for a tag that may only be occasionally overridden when updating a metric.

The more I think about this, the more I think that the BasicTagFactory is probably not actually needed.  If someone using a DynamicId has a particular tag that they want to use they should just use the non-dynamic withTag method to specify a value.

Thoughts?